sw/airborne/modules/computer_vision/opticflow/visual_estimator.c

```diff
@@ -0,0 +1,309 @@
+    for (int j = i + 1; j < count; j++) {
+      // distance squared:
+      distance2 = (x[i] - x[j]) * (x[i] - x[j]) + (y[i] - y[j]) * (y[i] - y[j]);
+      if (distance2 < min_distance2) {
+        labelmin[i] = 1;
+      }
+    }
+  }
+
+  int count_fil = count;
+  for (int i = count - 1; i >= 0; i--) {
+    remove_point = 0;
+
+    if (labelmin[i]) {
+      remove_point = 1;
+    }
+
+    if (remove_point) {
+      for (c = i; c < count_fil - 1; c++) {
+        x[c] = x[c + 1];
+        y[c] = y[c + 1];
+      }
+      count_fil--;
+    }
+  }
+
+  if (count_fil > max_count) { count_fil = max_count; }
+  count = count_fil;
+  free(labelmin);
+
+  // **********************************************************************************************************************
+  // Corner Tracking
+  // **********************************************************************************************************************
+  CvtYUYV2Gray(gray_frame, frame, imgWidth, imgHeight);
+
+  error_opticflow = opticFlowLK(gray_frame, prev_gray_frame, x, y, count_fil, imgWidth, imgHeight, new_x, new_y, status,
+                                5, 100);
+
+  flow_count = count_fil;
+  for (int i = count_fil - 1; i >= 0; i--) {
+    remove_point = 1;
+
+    if (status[i] && !(new_x[i] < borderx || new_x[i] > (imgWidth - 1 - borderx) ||
+                       new_y[i] < bordery || new_y[i] > (imgHeight - 1 - bordery))) {
+      remove_point = 0;
+    }
+
+    if (remove_point) {
+      for (c = i; c < flow_count - 1; c++) {
+        x[c] = x[c + 1];
+        y[c] = y[c + 1];
+        new_x[c] = new_x[c + 1];
+        new_y[c] = new_y[c + 1];
+      }
+      flow_count--;
+    }
+  }
+
+  dx_sum = 0.0;
+  dy_sum = 0.0;
+
+  // Optical Flow Computation
+  for (int i = 0; i < flow_count; i++) {
+    dx[i] = new_x[i] - x[i];
+    dy[i] = new_y[i] - y[i];
+  }
+
+  // Median Filter
+  if (flow_count) {
+    quick_sort_int(dx, flow_count); // 11
+    quick_sort_int(dy, flow_count); // 11
+
+    dx_sum = (float) dx[flow_count / 2];
+    dy_sum = (float) dy[flow_count / 2];
+  } else {
+    dx_sum = 0.0;
+    dy_sum = 0.0;
+  }
+
+  // Flow Derotation
+  curr_pitch = stateGetNedToBodyEulers_f()->theta;
+  curr_roll = stateGetNedToBodyEulers_f()->phi;
+  curr_yaw = stateGetNedToBodyEulers_f()->psi;
+
+  diff_pitch = (curr_pitch - prev_pitch) * imgHeight / FOV_H;
+  diff_roll = (curr_roll - prev_roll) * imgWidth / FOV_W;
+
+  prev_pitch = curr_pitch;
+  prev_roll = curr_roll;
+
+#ifdef FLOW_DEROTATION
+  if (flow_count) {
+    OFx_trans = dx_sum - diff_roll;
+    OFy_trans = dy_sum - diff_pitch;
+
+    if ((OFx_trans <= 0) != (dx_sum <= 0)) {
+      OFx_trans = 0;
+      OFy_trans = 0;
+    }
+  } else {
+    OFx_trans = dx_sum;
+    OFy_trans = dy_sum;
+  }
+#else
+  OFx_trans = dx_sum;
+  OFy_trans = dy_sum;
+#endif
+
+  // Average Filter
+  OFfilter(&OFx, &OFy, OFx_trans, OFy_trans, flow_count, 1);
+
+  // Velocity Computation
+#ifdef USE_SONAR
```

Comment:
USE_x flags should be checked with `#if USE_x` and not `#ifdef USE_x`.
Probably best to subscribe to the AGL ABI message to get the scale.